src/model/add-dots.test.ts

```diff
@@ -0,0 +1,131 @@
+import { enhancedDots } from './add-dots';
+import { bodyJSON } from './exampleBodyJSON';
+
+const example = JSON.parse(bodyJSON);
+
+describe('Replacing Middot with fake dot', () => {
+	it('creates an identical but new object when no changes are needed', () => {
+		expect(enhancedDots(example)).not.toBe(example); // We created a new object
+		expect(enhancedDots(example)).toEqual(example); // The new object is what we expect
+	});
+
+	it('replace the Middot', () => {
+		const input = {
+			...example,
+			blocks: [
+				{
+					elements: [
+						{
+							_type:
+								'model.dotcomrendering.pageElements.TextBlockElement',
+							html: '<p>I am the first paragraph</p>',
+						},
+						{
+							_type:
+								'model.dotcomrendering.pageElements.TextBlockElement',
+							html: '<p>* * *</p>',
+						},
+						{
+							_type:
+								'model.dotcomrendering.pageElements.TextBlockElement',
+							html: '<p>I should become a drop cap.</p>',
+						},
+						{
+							_type:
+								'model.dotcomrendering.pageElements.TextBlockElement',
+							html: '<p>• I should have a dot.</p>',
+						},
+					],
+				},
+			],
+		};
+
+		const expectedOutput = {
+			...example,
+			blocks: [
+				{
+					elements: [
+						{
+							_type:
+								'model.dotcomrendering.pageElements.TextBlockElement',
+							html: '<p>I am the first paragraph</p>',
+						},
+						{
+							_type:
+								'model.dotcomrendering.pageElements.TextBlockElement',
+							html: '<p>* * *</p>',
+						},
+						{
+							_type:
+								'model.dotcomrendering.pageElements.TextBlockElement',
+							html: '<p>I should become a drop cap.</p>',
+						},
+						{
+							_type:
+								'model.dotcomrendering.pageElements.TextBlockElement',
+							html:
+								'<p><span data-dcr-style="bullet"></span> I should NOT have a dot.</p>',
```

Comment:
It looks like the text after the dot has changed here which _should_ fail the test 🤔

Reply:
I've done the test as a 'not to be equal', so the input is not equal to the output, therefore correct.

Reply:
🤔 Sorry to nag but the test is called `it('replace the Middot'` and the expected output says

```
'<p><span data-dcr-style="bullet"></span> I should NOT have a dot.</p>',
```

But it does have a dot so it's not super clear what this test is proving

Reply:
One of the useful aspects of tests is they can act as a guide to what the code should do, so they form a sort of document for future developers to read and understand your intentions. For the benefit of these good people coming after us (and because these people may very well be _us_ after we've long forgotten about all this) we want these tests to be as clear and simple as possible